### PR TITLE
EZEE-3350: Added support to save additional data in ezImage FT

### PR DESCRIPTION
**Pull request by ViniTou**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | EZEE-3350
| **Type**                                   | feature
| **Target eZ Platform version** | `v3.3`
| **BC breaks**                          | no
| **Doc needed**                       | yes/no

As a part of image editor story, ezimage type must be able to save additional information with some related parameters. This opened up possibility to make it more versatile (and easier to maintain).

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/php-dev-team`).

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+use eZ\Publish\Core\IO\IOServiceInterface;
+use eZ\Publish\Core\IO\UrlRedecoratorInterface;
+use eZ\Publish\Core\IO\Values\BinaryFile;
+use eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\ImageConverter;
+use eZ\Publish\Core\Persistence\Legacy\Content\StorageFieldValue;
+use eZ\Publish\SPI\Persistence\Content\FieldValue;
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\PhpUnit\ClockMock;
+
+class ImageConverterTest extends TestCase
```
Comment (adamwojs) [inline eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php:20]:
```suggestion
final class ImageConverterTest extends TestCase
```

**Review by adamwojs — APPROVED** [review]

File: eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+        $expectedXml = <<< XML
+<?xml version="1.0" encoding="utf-8"?>
+<ezimage serial_number="1" is_valid="1" filename="ibexa_fav.png"
+    suffix="png" basename="ibexa_fav" dirpath="{$dir}" url="{$pathToImg}"
+    original_filename="ibexa_fav.png" mime_type="image/png" width="100"
+    height="200" alternative_text="test" alias_key="1293033771" timestamp="{$time}">
+  <original attribute_id="1" attribute_version="1" attribute_language="eng-GB"/>
+  <information Height="200" Width="100" IsColor="1"/>
+  <additional_data focalPointX="50" focalPointY="100" author="John Smith"/>
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php:59]: Are those data attributes/options dynamic (source system dependent and different based on a system)?
If so, then this XML should rather be something like:
```xml
<additional_data>
  <option key="focalPointX">50</option>
  <option key="focalPointY">100</option>
  <option key="author">John Smith</option>
</additional_data>
```
Dynamic values should never be neither XML tags nor attributes, so XSD schema can be derived easily.
Note that I took "option" from `buildAdditionalDataTag`, but it seems those are attributes? Maybe then it should be `attributes > attribute` instead?

File: composer.json [hunk 3]
```diff
@@ -63,7 +63,8 @@
         "friendsofphp/php-cs-fixer": "^2.16.2",
         "ezsystems/ezplatform-code-style": "^0.1",
         "phpunit/phpunit": "^8.2",
-        "matthiasnoback/symfony-dependency-injection-test": "^4.1"
+        "matthiasnoback/symfony-dependency-injection-test": "^4.1",
+        "symfony/phpunit-bridge": "^5.1"
```
Comment (alongosz) [inline composer.json:67]: How is this related to the PR?

Comment (ViniTou) [reply composer.json:67]: I use `ClockMock` for `time()` mocking in tests.

File: eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php (hunk 2 above)
Comment (ViniTou) [reply eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php]: I redone it in https://github.com/ezsystems/ezplatform-kernel/pull/137/commits/1429f5b67ba405853fd73c3b0b8baf98f9affef2

File: eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+<?xml version="1.0" encoding="utf-8"?>
+<ezimage serial_number="1" is_valid="1" filename="ibexa_fav.png"
+    suffix="png" basename="ibexa_fav" dirpath="{$dir}" url="{$pathToImg}"
+    original_filename="ibexa_fav.png" mime_type="image/png" width="100"
+    height="200" alternative_text="test" alias_key="1293033771" timestamp="{timestampToReplace}">
+  <original attribute_id="1" attribute_version="1" attribute_language="eng-GB"/>
+  <information Height="200" Width="100" IsColor="1"/>
+  <additional_data><attribute key="focalPointX">50</attribute><attribute key="focalPointY">100</attribute><attribute key="author">John Smith</attribute></additional_data>
+</ezimage>
+XML,
```
Comment (Steveb-p) [inline eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php:105]: https://wiki.php.net/rfc/flexible_heredoc_nowdoc_syntaxes

You can indent this, since we have PHP 7.3+ as dependency.

Comment (ViniTou) [reply eZ/Publish/Core/Persistence/Tests/Legacy/FieldValue/Converter/ImageConverterTest.php:105]: I also thought we can do this, but it seems that indenting XMLs causes issues with linters as XML must start with header and not empty spaces.

**Review by alongosz — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on Ibexa DXP 3.3.0-dev with diff.
